packages/core/core/src/core-api/routes/validation/attributes.ts

```diff
@@ -0,0 +1,641 @@
+ * appropriate Zod validation schemas with their respective constraints.
+ *
+ * @param attribute - The Strapi attribute configuration object. Contains type information
+ *                   and validation rules for the attribute.
+ *
+ * @returns A Zod schema that corresponds to the input attribute's type and validation rules
+ *
+ * @example
+ * ```typescript
+ * // String attribute with constraints
+ * const stringAttribute = {
+ *   type: 'string',
+ *   minLength: 3,
+ *   maxLength: 50,
+ *   required: true
+ * };
+ * const stringSchema = mapAttributeToSchema(stringAttribute);
+ *
+ * // Enumeration attribute
+ * const enumAttribute = {
+ *   type: 'enumeration',
+ *   enum: ['draft', 'published', 'archived']
+ * };
+ * const enumSchema = mapAttributeToSchema(enumAttribute);
+ *
+ * // Media attribute with multiple files
+ * const mediaAttribute = {
+ *   type: 'media',
+ *   multiple: true
+ * };
+ * const mediaSchema = mapAttributeToSchema(mediaAttribute);
+ * ```
+ *
+ * @throws {Error} Throws an error if an unsupported attribute type is provided
+ *
+ * @remarks
+ * - Complex types (component, relation, dynamic zone) use placeholder schemas
+ *
+ * @todo: Move to @strapi/utils if needed for other validation
+ */
+export const mapAttributeToSchema = (attribute: Schema.Attribute.AnyAttribute): z.Schema => {
+  switch (attribute.type) {
+    case 'biginteger': {
+      const { writable, required, min, max, default: defaultValue } = attribute;
+
+      const schema = augmentSchema(z.string(), [
+        (schema) => (min !== undefined ? schema.min(min as unknown as number) : schema),
+        (schema) => (max !== undefined ? schema.max(max as unknown as number) : schema),
+        maybeRequired(required),
+        maybeWithDefault(defaultValue),
+        maybeReadonly(writable),
+      ]);
+
+      return schema.describe('A biginteger field');
+    }
+    case 'blocks': {
+      // TODO: better support blocks data structure
+      return z.array(z.any()).describe('A blocks field');
+    }
+    case 'boolean': {
+      const { writable, required, default: defaultValue } = attribute;
+
+      const schema = augmentSchema(z.boolean(), [
+        maybeRequired(required),
+        maybeWithDefault(defaultValue),
+        maybeReadonly(writable),
+      ]);
+
+      return schema.describe('A boolean field');
+    }
+    case 'component': {
+      const { _idmap } = z.globalRegistry;
+
+      const set = (id: string, schema: z.ZodType) => {
+        if (_idmap.has(id)) {
+          _idmap.delete(id);
+        }
+
+        z.globalRegistry.add(schema, { id });
+      };
+
+      const { writable, required, min, max, component, repeatable } = attribute;
+
+      const existsInGlobalRegistry = _idmap.has(component);
+
+      let componentSchema: z.ZodType;
+
+      if (existsInGlobalRegistry) {
+        componentSchema = _idmap.get(component) as z.ZodType;
+      } else {
+        set(component, z.any());
```

Comment:
Maybe we should comment here that this is what prevents the infinite loop? I don't think this is very easy to follow